### PR TITLE
Fix Meta.base_manager_name check with multiple inheritance

### LINKED ISSUE #228: Check does not work with multiple base classes.
### Python Version

3.10

### Django Version

3.2.17

### Package Version

1.5.0

### Description

We recently upgraded to a newer version of `django-auto-prefetch` and found out that the check `_check_meta_inheritance` fails with "elaborate" model inheritance:

https://github.com/tolomea/django-auto-prefetch/blob/05d4f2a98a1f5e3f8ab0db5efac2e4c0e73faf42/src/auto_prefetch/__init__.py#L140-L144

The issue is that `cls.Meta` is not always the same class as the one in the code. I believe it's due to these two snippets in Django:

* https://github.com/django/django/blob/c914d6cff176ae6bfab2f33a84bcfd45208f1894/django/db/models/base.py#L110
* https://github.com/django/django/blob/c914d6cff176ae6bfab2f33a84bcfd45208f1894/django/db/models/base.py#L357-L363

TL;DR: if the `Meta.abstract = False` Django effectively drops it in the `ModelBase.__new__`. As a result, the first parent class's `Meta` is inherited. For example, to reproduce:

```python
# foo/models.py
from django.db import models
from django.contrib.auth.models import PermissionsMixin
import auto_prefetch


class Foo(PermissionsMixin, auto_prefetch.Model, models.Model):
    class Meta(PermissionsMixin.Meta, auto_prefetch.Model.Meta):
        app_label = "foo"

    id = models.PositiveIntegerField()

# python manage.py shell
from foo.models import Foo
import auto_prefetch

print(issubclass(Foo.Meta, auto_prefetch.Model.Meta))
# Output: False
```

Incidentally, workaround is not hard. In our codebase we retain the original `_Meta` using an extended `ModelBase`:

```python
class ModelBase(models.base.ModelBase):
    def __new__(cls, name, bases, attrs, **kwargs):
        # preserve the actual Meta as _Meta since Django deletes it for concrete models
        if "Meta" in attrs:
            attrs["_Meta"] = attrs["Meta"]
        return super().__new__(cls, name, bases, attrs)
```

I'm not sure if this counts as Django bug or `auto_prefetch`. Happy to have a go at `auto_prefetch` PR later this week, as I think there is a way to recover `Meta` from `instance._meta`, but I have not looked into it.

### PR DESCRIPTION
**Pull request by adamchainz**
Fixes #228.

### CODE REVIEW

**Comment by seporaitis** [comment]
Installed the code from this branch and ran `manage.py check` - can confirm the error is gone.

**Comment by adamchainz** [comment]
Released in 1.5.1.

**Comment by seporaitis** [comment]
Thanks! We are already using it. :)
